include/libdnf/base/base.hpp

```diff
@@ -71,6 +72,10 @@ class Base {
     /// Returns a pointer to a locked "Base" instance or "nullptr" if no instance is locked.
     static Base * get_locked_base() noexcept;
 
+    /// Call a function that loads the config file, catching errors appropriately
+    template <typename T>
+    T with_config_file_path(std::function<T(const std::string &)> func);
```

Comment:
Is making the method templated needed here? Both current usages are void functions with the same signature, do you think of any different use case now? Because, IMHO, it distracts the user and adds some unnecessary complexity there...

Reply:
Sure, I'll take out the generic and just make it take a `void` function.